pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go

```diff
@@ -174,3 +176,103 @@ func aggregateJobStatus(object *unstructured.Unstructured, aggregatedStatusItems
+		klog.V(3).Infof("Grab daemonSet(%s/%s) status from cluster(%s), currentNumberScheduled: %d, desiredNumberScheduled: %d, numberAvailable: %d, numberMisscheduled: %d, numberReady: %d, updatedNumberScheduled: %d",
+			daemonSet.Namespace, daemonSet.Name, item.ClusterName, temp.CurrentNumberScheduled, temp.DesiredNumberScheduled, temp.NumberAvailable, temp.NumberMisscheduled, temp.NumberReady, temp.UpdatedNumberScheduled)
+		newStatus.ObservedGeneration = daemonSet.Generation
+		newStatus.CurrentNumberScheduled += temp.CurrentNumberScheduled
+		newStatus.DesiredNumberScheduled += temp.DesiredNumberScheduled
+		newStatus.NumberAvailable += temp.NumberAvailable
+		newStatus.NumberMisscheduled += temp.NumberMisscheduled
+		newStatus.NumberReady += temp.NumberReady
+		newStatus.UpdatedNumberScheduled += temp.UpdatedNumberScheduled
+	}
+
+	if oldStatus.ObservedGeneration == newStatus.ObservedGeneration &&
+		oldStatus.CurrentNumberScheduled == newStatus.CurrentNumberScheduled &&
+		oldStatus.DesiredNumberScheduled == newStatus.DesiredNumberScheduled &&
+		oldStatus.NumberAvailable == newStatus.NumberAvailable &&
+		oldStatus.NumberMisscheduled == newStatus.NumberMisscheduled &&
+		oldStatus.NumberReady == newStatus.NumberReady &&
+		oldStatus.UpdatedNumberScheduled == newStatus.UpdatedNumberScheduled {
+		klog.V(3).Infof("ignore update daemonSet(%s/%s) status as up to date", daemonSet.Namespace, daemonSet.Name)
+		return object, nil
+	}
+
+	oldStatus.ObservedGeneration = newStatus.ObservedGeneration
+	oldStatus.CurrentNumberScheduled = newStatus.CurrentNumberScheduled
+	oldStatus.DesiredNumberScheduled = newStatus.DesiredNumberScheduled
+	oldStatus.NumberAvailable = newStatus.NumberAvailable
+	oldStatus.NumberMisscheduled = newStatus.NumberMisscheduled
+	oldStatus.NumberReady = newStatus.NumberReady
+	oldStatus.UpdatedNumberScheduled = newStatus.UpdatedNumberScheduled
+
+	return helper.ToUnstructured(daemonSet)
+}
+
+func aggregateStatefulSetStatus(object *unstructured.Unstructured, aggregatedStatusItems []workv1alpha2.AggregatedStatusItem) (*unstructured.Unstructured, error) {
+	statefulSet, err := helper.ConvertToStatefulSet(object)
+	if err != nil {
+		return nil, err
+	}
+	oldStatus := &statefulSet.Status
+	newStatus := &appsv1.StatefulSetStatus{}
+	for _, item := range aggregatedStatusItems {
+		if item.Status == nil {
+			continue
+		}
+		temp := &appsv1.StatefulSetStatus{}
+		if err = json.Unmarshal(item.Status.Raw, temp); err != nil {
+			return nil, err
+		}
+		klog.V(3).Infof("Grab statefulSet(%s/%s) status from cluster(%s), availableReplicas: %d, currentReplicas: %d, readyReplicas: %d, replicas: %d, updatedReplicas: %d",
+			statefulSet.Namespace, statefulSet.Name, item.ClusterName, temp.AvailableReplicas, temp.CurrentReplicas, temp.ReadyReplicas, temp.Replicas, temp.UpdatedReplicas)
+		newStatus.ObservedGeneration = statefulSet.Generation
+		newStatus.CurrentRevision = temp.CurrentRevision
+		newStatus.UpdateRevision = temp.UpdateRevision
```

Comment:
Those two fields maybe need not to calculate.

Reply:
Actually I consider if need to delete them. It seems deleting is better.